File: program29.c
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function name        : RangesumEven 
// Input                : Integer
// Output               : Integer, Integer
// Description          : It is use to return the sum of even numbers between the given range
// Author               : Vishal Tejkant Patil
// Date                 : 16/09/2021
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
#include<stdio.h>

int RangeSumEven(int iStart, int iEnd)
{
        int iCnt=0,iSum=0;
        if (iStart>iEnd || iStart<0)
        {
                printf("Invalid Input\n");
        }
        else
        {
                for(iCnt=iStart;iCnt<=iEnd;iCnt++)
                {
			if(iCnt%2==0)
			{	
                        	iSum = iSum+iCnt;
			}	
                }
        }

        return iSum;
}

int main()
{
        int iValue1=0,iValue2=0,iRet=0;

        printf("Enter starting point ");
        scanf("%d",&iValue1);

        printf("Enter ending point ");
        scanf("%d",&iValue2);

        iRet = RangeSumEven(iValue1,iValue2);
        if(iRet!=0)
        {
                printf("Addition is %d\n",iRet);
        }

        return 0;
}


////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	OUTPUT:
//
//		Enter starting point 23
//		Enter ending point 30
//		Addition is 108
//
//		Enter starting point 10
//		Enter ending point 18
//		Addition is 70
//
//		Enter starting point -10
//		Enter ending point 2
//		Invalid Input
//
//		Enter starting point 90
//		Enter ending point 18
//		Invalid Input
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program29.java
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept range from user and return addition of all even numbers in between that range. 
//		(Range should contains positive numbers only)
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.*;

class Solution
{
	public int iStart,iEnd;

	public Solution(int iStart,int iEnd)
	{
		this.iStart = iStart;
		this.iEnd = iEnd;
	}

	public int RangeEvenSum()
	{

  		int iSum=0;		
		if(iStart<0 || iEnd<0)
		{
			System.out.println("Invalid Syntax");
			System.exit(0);
		}		

		for(int iCnt=iStart;iCnt<=iEnd;iCnt++)
		{
			if(iCnt%2==0)
			{
				iSum=iSum+iCnt;
			}	
		}

		return iSum;
	
	}
}

class Program4
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		int iRet=0;
		System.out.println("Enter Start: ");
		int iStart = sObj.nextInt();
		
		System.out.println("Enter End: ");
		int iEnd = sObj.nextInt();

		Solution obj = new Solution(iStart,iEnd);
		iRet = obj.RangeEvenSum();
	
		System.out.println("Addition of Even in the range is: "+iRet);
	}
}

/*
OUTPUT:

Enter Start: 
23
Enter End: 
30
Addition of Even in the range is: 108

*/
